newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/UsersAutocomplete.js

```diff
@@ -0,0 +1,149 @@
+//@flow
+import React from 'react';
+import { useState } from 'react';
+import { t } from '@lingui/macro';
+import SemiControlledMultiAutoComplete from '../../UI/SemiControlledMultiAutoComplete';
+import {
+  searchUserPublicProfilesByUsername,
+  type UserPublicProfile,
+  getUserPublicProfilesByIds,
+} from '../../Utils/GDevelopServices/User';
+
+import useForceUpdate from '../../Utils/UseForceUpdate';
+
+type Props = {|
+  eventsFunctionsExtension: gdEventsFunctionsExtension,
```

Comment:
I think we should directly make this a generic component that takes not a eventsFunctionsExtension, but a gdVectorString :D 

`userIds: gdVectorString`

(note that I put `userIds` and not `users` to make it clear what this contains. It's the equivalent of the name of the field in the extension "authorIds").
You then just have to pass it as prop `<UsersAutocomplete userIds={eventsFunctionsExtension.getAuthorIds()} />` (which make it clear that we're editing the users that are the authors of the extension)

We can then change all reference to `author` to `user` in this component and it will work exactly the same.

Reply:
Yes!